dask/bag/core.py

```diff
@@ -297,18 +299,33 @@ def remove(self, predicate):
                    for i in range(self.npartitions))
         return type(self)(merge(self.dask, dsk), name, self.npartitions)
 
-    def map_partitions(self, func):
+    def map_partitions(self, func, *args):
         """ Apply function to every partition within collection
 
         Note that this requires you to understand how dask.bag partitions your
         data and so is somewhat internal.
 
         >>> b.map_partitions(myfunc)  # doctest: +SKIP
+
+        Any additional arguments get passed to the function _after_ the
+        partition argument; argument values are first passed through
+        dask.imperative.to_task_dasks and any resulting dasks merged into the
+        result Bag dask:
+
+        >>> b.map_partitions(lambda part, freqs: part / freqs, b.frequencies())  # doctest: +SKIP
```

Comment:
Note that this will have some negative consequences when it comes to memory usage.  We'll end up keeping a fair amount more in RAM than otherwise.